controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -434,12 +434,10 @@ public OperationContext createContext(String scope, String name) {
                                                        final Executor executor) {
         CompletableFuture<Void> future = withCompletion(getStream(scope, name, context).scaleOldSegmentsSealed(sealedSegmentSizes, record), executor);
 
-        future.thenCompose(result -> CompletableFuture.allOf(
+        future.thenCompose(result -> 
                 getActiveSegments(scope, name, null, executor).thenAccept(list ->
-                        StreamMetrics.reportActiveSegments(scope, name, list.size())),
-                findNumSplitsMerges(scope, name, executor).thenAccept(simpleEntry ->
```

Comment:
This is removing the reporting of splits and merges, but it is not listed in the PR description. Is there any real impact to the reporting of splits and merges the fact that we aren't doing on every scale call. In particular, I'm wondering about this issue #1761 .

Reply:
There are two very bad metrics computation --
1. splits and merges
2. count open transactions
both are costly and scan metadata. 

Splits and merges can be stored in the metadata and updated with every scale to make it efficient. 

Count open transaction: since transactions are getting created and committed/aborted continuously, the open transaction count is very costly as it reads metadata for all transactions and then filters those that are open..
since create transaction request can be sent to any controller instance, so there can be no coordinated count. 
keeping a common counter in metadata will also be detrimental as it will need to be updated on each of the above mentioned operation (create/commit/abort).

Reply:
The confusing part is what is the one on what is happening to the metrics. You need to clarify in the description whether the optimization has any impact to the consumer of metrics.

Reply:
Yes, i have removed the metrics update altogether. we dont want to continue computing these metrics whose computation is typically more expensive than a workflow execution. 

For scale we have an explicit api which the UI makes use of. 
number of open transactions is something we can add a new api or do something else. But in its current form it is not sustainable, esp with our kvs implementation where this costly computation hammers segmentstore and has detrimental effect on segment store's performance.